Turbulence.ModelGenerator/Preprocessing.cs

```diff
@@ -0,0 +1,136 @@
+        List<string> writtenFiles = new();
+
+        foreach (string file in Directory.EnumerateFiles(inPath.AbsolutePath, "*.*", SearchOption.AllDirectories))
+        {
+            using StreamReader reader = new(file);
+
+            // Keep track of the last ## header found, because it will be used to name JSON Params, JSON/Form and Query String Params tables
+            string? lastH2 = null;
+
+            string? line;
+            while ((line = await reader.ReadLineAsync()) != null)
+            {
+                if (line.StartsWith("## "))
+                {
+                    lastH2 = line;
+                }
+                
+                // Based on the assumption that every table has H6 above it, which it should according to the README
+                if (!line.StartsWith("###### ")) continue;
+
+                string? name;
+                
+                /* language=RegEx */
+                const string paramsPattern = @"###### (.*?)(Query String Params|JSON Params|JSON/Form Params|Response Body)$";
+                var match = Regex.Match(line, paramsPattern);
+                if (match.Success && lastH2 != null)
+                {
+                    // Use the last header title instead of [...] Params
+                    name = Regex.Match(lastH2, @"## (.*) %.*").Groups[1].Value;
+
+                    if (line.Contains("Query String Params"))
+                    {
+                        name += "QueryParams";
+                    }
+                    else if (line.Contains("Response Body"))
+                    {
+                        name += "Response";
+                    }
+                    else
+                    {
+                        name += "Params";
+                    }
+                }
+                else
+                {
+                    // Markdown heading to pretty class name
+                    /* language=RegEx */
+                    const string pattern = @"###### (.*?)( Structure| Fields)?$";
+                    name = Regex.Match(line, pattern).Groups[1].Value;
+                }
+
+                name = name.Replace(" ", "")
+                           .Replace("/", "")
+                           .Replace("-", "");
+
+                string? table = await ExtractTable(reader);
+
+                // Not a valid table or not a table we are interested in, continue
+                if (table == null) continue;
+
+                // We use the name of the file for the directory name to store all the tables to
+                // We assume there are no duplicates
+                string dir = Regex.Replace(file, ".md$", "")
+                                  .Split(Path.DirectorySeparatorChar)[^1]
+                                  .Replace('_', ' ');
+                dir = CultureInfo.InvariantCulture.TextInfo.ToTitleCase(dir).Replace(" ", "");
+                
+                // Special case for OAuth2
+                if (dir == "Oauth2") dir = "OAuth2";
```

Comment:
Ouchie

Reply:
I guess I could try to leave the original uppercase characters intact?

Reply:
Just feels like a separate data case in code is yucky